### ISSUE TITLE
Consider using duckdb

### ISSUE THREAD

**Issue by domoritz** [spacegraphcats/spacegraphcats#339]
https://duckdb.org/ is a database with almost the same API as SQLite but it uses a column store and has generally optimizations for analytical queries. We could swap it out to see if it helps with queries. 

Especially max queries like https://github.com/spacegraphcats/spacegraphcats/blob/6394904098591ad2f9e6c000212ebb42237577c0/spacegraphcats/search/search_utils.py#L13 should be much faster.

**Comment by domoritz** [comment]
Alternatively, we could consider Arrow/Parquet (https://arrow.apache.org/docs/python/parquet.html) as the storage format.

**Comment by taylorreiter** [comment]
ooh and arrow/parquet is R compatible as well. I've noticed the file format is lightning fast, but sometimes takes up a lot of disk space. I'm not sure how it compares to duckdb though.

**Comment by ctb** [comment]
right, and #345 will set things up so that it's straightforward to swap out backends and make sure that results are identical across versions. I will try to spell out some basic benchmarking options, too!

**Comment by ctb** [comment]
we are in the interesting position where after creation this database is read-only. So there might be some great options out there that improve both creation and query!
